src/importexport/musicxml/internal/musicxml/exportxml.cpp

```diff
@@ -4281,13 +4281,37 @@ void ExportMusicXml::rest(Rest* rest, staff_idx_t staff, const std::vector<Lyric
     // as no display-step or display-octave should be written for a tablature staff,
 
     if (clef != ClefType::TAB && clef != ClefType::TAB_SERIF && clef != ClefType::TAB4 && clef != ClefType::TAB4_SERIF) {
-        double yOffsSp = rest->offset().y() / rest->spatium();                  // y offset in spatium (negative = up)
-        yOffsSt = -2 * int(yOffsSp > 0.0 ? yOffsSp + 0.5 : yOffsSp - 0.5);     // same rounded to int (positive = up)
+        double yOffsSp = -2 * rest->offset().y() / rest->spatium();              // positive = up, one spatium is two pitches
+        yOffsSt = int(yOffsSp > 0.0 ? yOffsSp + 0.5 : yOffsSp - 0.5);            // same rounded to int
 
         po -= 4;        // pitch middle staff line (two lines times two steps lower than top line)
-        po += yOffsSt;     // rest "pitch"
-        oct = po / 7;     // octave
-        stp = po % 7;     // step
+        po += yOffsSt;  // rest "pitch"
+        po -= 7;        // correct octave
+        // at this point:
+        // C0 (lowest value allowed in MusicXML) has po == 0   (y offset = 17 (treble clef) 11 (bass clef))
+        // B9 (highest value allowed in MusicXML) has po == 69 (y offset = -17.5 (treble clef) -23.5 (bass clef))
+
+        if (po < 0) {
+            po = 0;     // apply limits
+        }
+        if (po > 69) {
+            po = 69;    // apply limits
+        }
```

Comment:
This looks like a good use case for `std::clamp`:
```suggestion
        po = std::clamp(po, 0, 69);
```

Reply:
Requires C++17 though